django/db/models/sql/query.py

```diff
@@ -890,18 +897,25 @@ def join(self, join, reuse=None):
         The 'reuse' parameter can be either None which means all joins are
         reusable, or it can be a set containing the aliases that can be reused.
 
+        The 'force_reuse' parameter is the same type of 'reuse' but with difference
+        that it bypasses the 'reuse' paramter. It is used when computing.
```

Comment:
I don't see why a new arg is needed. We could just accept `True` for `reuse` and make that force it.... `if isinstance(reuse, bool) and reuse:`

Also the "It is used when computing" sentence seems incomplete. We're *always* computing 😉

Reply:
If given, `reuse` (and `force_reuse`) is expected to be a dictionary.
So `isinstance(reuse, bool)` will never be `True`.

Reply:
Good point, I misunderstood. That said it seems that `force_reuse` could really just be a flag like `reuse_ with_filtered_relation` which just determines whether `with_filtered_relation` is passed to `join.equals` ?

Reply:
I will give it a try

Reply:
in 17a3e3ee2f9d1553a48a98a10d4c52ec507fe4ae
What do you think ?